dustcloud/server.py

```diff
@@ -31,528 +28,527 @@
+            return
+        else:
+            if data[0:2] == bytes.fromhex("2131"):  # Check for magic bytes
+                print("magic ok")
+                packetlenght = int.from_bytes(data[2:4], byteorder='big')  # get packet lenght from header
+                # print("packetlength %s, got already %s " % (packetlenght,len(data)))
+                while len(data) != packetlenght:  # packet longer than 32 byte
+                    data += self.request.recv((packetlenght - 32))  # get the rest of the packet
+                # print("packetlength %s, got now %s " % (packetlenght,len(data)))
+                # print("= RAW: %s" % binascii.hexlify(data))
+                process_result = self.Cloudi.process_data(self, data)
+                if process_result == 1:
+                    self.request.close()
+                    return
+            else:
+                data += self.request.recv(64*1024)  # get the rest of the packet
+                print("Unknown message: {}".format(data))
+                return
+
+
 class MyUDPHandler(socketserver.BaseRequestHandler):
-	"""
-	This class works similar to the TCP handler class, except that
-	self.request consists of a pair of data and client socket, and since
-	there is no connection the client address must be given explicitly
-	when sending data back via sendto().
-	"""
-	clienthello = ""
-	ddid = 0
-	ctx = ""
-	device_id = ""
-	dname = ""
-	sended = 0
-	commandcounter = 0
-	lastmessageid = 0
-	connmode= "udp"
-	
-	def sendmydata(self,data):
-		self.request.sendto(data, self.client_address)
-
-	def handle(self):
-		data = self.request[0].strip()
-		socket = self.request[1]
-		print("{} via udp wrote:".format(self.client_address[0]))
-		# print("> RAW: %s" % binascii.hexlify(data))
-		if len(data) < 32:
-			print("len < 32")
-			return
-		else:
-			if data[0:2] == bytes.fromhex("2131"):  # Check for magic bytes
-				print("magic ok")
-				process_data(socket,data)
+    """
+    This class works similar to the TCP handler class, except that
+    self.request consists of a pair of data and client socket, and since
+    there is no connection the client address must be given explicitly
+    when sending data back via sendto().
+    """
+    clienthello = ""
+    ddid = 0
+    ctx = ""
+    device_id = ""
+    dname = ""
+    sended = 0
+    commandcounter = 0
+    lastmessageid = 0
+    connmode = "udp"
+
+    def sendmydata(self, data):
+        self.request.sendto(data, self.client_address)
+
+    def handle(self):
+        data = self.request[0].strip()
+        socket = self.request[1]
+        print("{} via udp wrote:".format(self.client_address[0]))
+        # print("> RAW: %s" % binascii.hexlify(data))
+        if len(data) < 32:
+            print("len < 32")
+            return
+        else:
+            if data[0:2] == bytes.fromhex("2131"):  # Check for magic bytes
+                print("magic ok")
+                # TODO: process data
+                # process_data(socket, data)

```

Comment:
MyUDPHandler doesn't have a process_data method, I guess this was supposed to also set up a CloudClient to process messages and store it in the database?